lib/fathead/scala/parse.py

```diff
@@ -0,0 +1,52 @@
+from bs4 import BeautifulSoup
+import os
+
+BASE_DIRECTORY = "./download/api/scala"
+BASE_URL = "https://www.scala-lang.org/api/current/"
+
+
+def get_all_files():
+	doc_files = []
+
+	for (path, dirs, files) in os.walk(BASE_DIRECTORY):
+		for f in files:
+			doc_files.append( "%s/%s" % (path, f))
+	return doc_files
+
+def get_docs(filename):
+	if filename[-5:] == ".html":
+		soup = BeautifulSoup(open(filename), 'html.parser')
+		title_link = soup.find_all("h1")[0]
+		title_text = title_link.text
+		
+		url = BASE_URL + filename.replace("./download/api/", "") #title_link.find("a").get('href')
+
+		article_text = soup.find_all("div", attrs={"class" : "comment"})[0].text
```

Comment:
For example: http://scala-lang.org/files/archive/nightly/docs/library/index.html#scala.Dynamic

We can have:

```html
<section>
    <p>{description text}</p>
    <pre>
        <code>{code}</code>
    </pre>
</section>
```